src/v/cluster/topics_frontend.cc

```diff
@@ -521,25 +526,30 @@ ss::future<topic_result> topics_frontend::replicate_create_topic(
     }
 
     return replicate_and_wait(_stm, _as, std::move(cmd), timeout)
-      .then_wrapped([tp_ns = std::move(tp_ns), units = std::move(units)](
-                      ss::future<std::error_code> f) mutable {
-          try {
-              auto error_code = f.get0();
-              auto ret_f = ss::now();
-              return ret_f.then(
-                [tp_ns = std::move(tp_ns), error_code]() mutable {
-                    return topic_result(std::move(tp_ns), map_errc(error_code));
-                });
+      .then_wrapped(
+        [tp_ns = std::move(tp_ns),
+         units = std::move(units),
+         &tt = _topics.local()](ss::future<std::error_code> f) mutable {
+            try {
+                auto error_code = f.get0();
+                auto ret_f = ss::now();
+                return ret_f.then(
+                  [tp_ns = std::move(tp_ns), error_code, &tt]() mutable {
+                      auto pc = tt.get_topic_partition_count(tp_ns);
```

Comment:
it occurred to me that this is possibly why the original author left this functionality out in the first place, a possible solution would be to just return the partition count and rf factor that was passed within the request. Because if the topic was created with success, then it must have been created with the parameters given within the request.

Reply:
> Replicate and wait...just returns when the command is replicated

Ah, don't think I fully grokked that

>this is possibly why the original author left this functionality out in the first place

yeah, that's what I'm thinking. put this code up basically so we can have this conversation.

>if the topic was created with success, then it must have been created with the parameters given

good point; I'm cool with this

Reply:
even better, I think we can just pull from the metadata cache right there in the handler? similar to how we grab configs and confirm that the topic exists. which is to say, nothing "new"

Reply:
Yeah right in the handler I see you've put that after the call to `wait_for_topics` LGTM